File: 백준/Silver/1002. 터렛/터렛.py
import sys
input = sys.stdin.readline
print = sys.stdout.write

def get_distance(x1,y1,x2,y2):
    return ((x2-x1)**2 + (y2-y1)**2)**0.5

test_case = int(input().rstrip("\n"))
for i in range(test_case):
    x1,y1,r1,x2,y2,r2 = map(int,input().split())
    distance = get_distance(x1,y1,x2,y2)
    if(r1+r2<distance):
        print("0\n")
    elif(r1+r2>distance):
        if(abs(distance)+r1 < r2 or abs(distance)+r2 < r1):
            print("0\n")
        elif(abs(distance)+r1 == r2 or abs(distance)+r2 == r1):
            if(x1==x2 and y1==y2):
                print("-1\n")
            else:
                print("1\n")
        else:
            print("2\n")
    else:
        if(x1==x2 and y1==y2):
            print("-1\n")
        else:
            print("1\n")

File: 백준/Silver/1002. 터렛/터렛.cc
#include<iostream>
#include<cmath>
using namespace std;

double get_distance(double x1, double y1, double x2, double y2) {
    return pow((pow((x2 - x1), 2) + pow((y2 - y1), 2)), 0.5);
}

int main() {
	int Test_case;
	cin >> Test_case;
	while (Test_case--) {
		double x1, x2, y1, y2, r1, r2;
		cin >> x1 >> y1 >> r1 >> x2 >> y2 >> r2;
        double distance = get_distance(x1, y1, x2, y2);
        if (r1 + r2 < distance) {
            cout << 0 << endl;
        }
        else if (double(r1 + r2) > distance) {
            if (abs(distance) + r1 < r2 || abs(distance) + r2 < r1)
                cout << 0 << endl;
            else if (abs(distance) + r1 == r2 || abs(distance) + r2 == r1) {
                if (x1 == x2 && y1 == y2) {
                    cout << -1 << endl;
                }
                else {
                    cout << 1 << endl;
                }
            }
            else {
                cout << 2 << endl;
            }
        }

        else {
            if (x1 == x2 && y1 == y2) {
                cout << -1 << endl;
            }
            else {
                cout << 1 << endl;
            }
        }
	}
    return 0;
}